### ISSUE TITLE
Replace search with Algolia based one

### ISSUE THREAD

**Issue by AuHau** [ethersphere/bee-docs#271]
Looking at other Docusaurus based documentation they almost always use the Algolia based search platform which is IMHO much better and nicer than the one we use. I would suggest replacing it.

https://docusaurus.io/docs/search#using-algolia-docsearch

**Comment by AuHau** [comment]
cc @ethersphere/js

**Comment by vojtechsimetka** [comment]
Good suggestion but I think we should have usability vs ideology discussion.

I was looking at Algolia when we started with the bee-js-docs. At that point I decided that we probably don't want to add a centralised solution. Especially if our goal is to host it (purely) on Swarm in future. IMHO the current search is good enough while working without 3rd party service.

**Comment by significance** [comment]
indeed this decision was made to avoid reliance on centralised services and so we can upload in a self contained way to swarm. @AuHau perhaps we can begin to discuss what you prefer about the Algolia search choice and we can try to implement it in the forthcoming design improvements.

**Comment by AuHau** [comment]
Yeah, I though that might be the case. Honestly, I don't really need Algolia per se, but I really love a few things that they have in their implementations:

 - the `cmd+k` / `ctrl+k` shortcut that focus the search bar (and the fact that a note about this is present in the search bar) 
<img width="171" alt="image" src="https://user-images.githubusercontent.com/6072250/171816661-abd803ae-93c3-40bb-93e3-fdaa9e46350e.png">
 - the modal search window (but I could get over that one)

**Comment by EchinopsisM** [comment]
## Proposed Closure

This issue from 2022-04-06 proposes replacing the search with Algolia. Given that:
- This is a **2+ year old** tooling decision
- The issue likely has been either implemented, rejected, or superseded
- No context about current status or reasoning

**I recommend closing this issue**. If the search implementation needs to be revisited, a new issue with current requirements and evaluation criteria should be opened instead.
